### ISSUE TITLE
call the ReduceLROnPlateau equivalent for Analytics Zoo when wrap keras_model within the Estimator.from_keras wrapper

### ISSUE THREAD

**Issue by jack1981** [intel-analytics/analytics-zoo#3060]
keras has this api (https://keras.io/api/callbacks/reduce_lr_on_plateau/).
callback exist in TF Keras, and also in BigDL. Using the callback to train is really simple for both cases, you just need to add this argument into the fit function.

For example in TF.keras:

keras_model = some.layers.etc
keras_model.fit(…., callbacks = [tf.keras.callbacks.ReduceLROnPlateau()])

But for Analytics Zoo, it’s:

est = Estimator.from_keras(keras_model, model_dir=log_dir)

# Train Model
logger.info("Start model training. Max epoch is set at" + args.max_epoch)
est.fit(data=train_df,
        batch_size=batch_size,
        epochs=max_epoch,
        feature_cols=['input_1'],
        labels_cols=['decoder_3', 'output'],
        validation_data=test_df))

There’s no way we can call the ReduceLROnPlateau equivalent for Analytics Zoo when we wrap our keras_model within the Estimator.from_keras wrapper.

**Comment by yangw1234** [comment]
Hi Jack，I have just merge a pr about this.

For a learning rate schedule that can be calculated deterministically before the training, I would recommend you to use the  tf.keras.optimizers.schedules package. You can refer to this code snippet for usage. https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/test/zoo/orca/learn/spark/test_estimator_keras_for_spark.py#L45

For ReduceLROnPlateau, which is calculated in runtime based on training results, it is kind of tricky for our current implementation. I have just merged a workaround. You can check out the following code for reference. https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/test/zoo/orca/learn/spark/test_estimator_keras_for_spark.py#L478

**Comment by helenlly** [comment]
@jack1981 any more questions on this issue?
